functests/gatekeeper/gatekeeper.go

```diff
@@ -0,0 +1,637 @@
+				value: "a",
+			},
+		}
+
+		By("Creating assignMetadata mutation-a")
+		amA := assignMetadataDefinition(amDefA)
+		amA, err = client.Resource("assignmetadata").Create(context.Background(), amA, metav1.CreateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		defer func() {
+			err := client.Resource("assignmetadata").Delete(context.Background(), amA.GetName(), metav1.DeleteOptions{})
+			Expect(err).ToNot(HaveOccurred())
+		}()
+
+		By("Creating test-pod-a")
+		testPodA := GetTestPod(TestingNamespace)
+		testPodA.SetName("test-pod-a")
+		testPodA.Unstructured, err = client.Resource("pods").Namespace(testPodA.GetNamespace()).Create(context.Background(), testPodA.Unstructured, metav1.CreateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		defer func() {
+			err := client.Resource("pods").Namespace(testPodA.GetNamespace()).Delete(context.Background(), testPodA.GetName(), metav1.DeleteOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+		}()
+
+		By("Asserting that test-pod-a was labeled")
+		value, ok, err = unstructured.NestedString(testPodA.Object, "metadata", amDefA.metadata.kind, amDefA.metadata.key)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(ok).To(BeTrue())
+		Expect(value).To(Equal(amDefA.metadata.value))
+	})
+
+	It("should be able to update mutation policy", func() {
+		amDef := &assignMetadataInfo{
+			name:  "mutation-version",
+			kinds: []string{"Pod"},
+			scope: "Namespaced",
+			metadata: metadataInfo{
+				kind:  "labels",
+				key:   "mutation-version",
+				value: "0",
+			},
+		}
+
+		am := assignMetadataDefinition(amDef)
+
+		By("Creating assignMetadata mutation-version with mutation-version: 0")
+		am = assignMetadataDefinition(amDef)
+		am, err := client.Resource("assignmetadata").Create(context.Background(), am, metav1.CreateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		defer func() {
+			err := client.Resource("assignmetadata").Delete(context.Background(), am.GetName(), metav1.DeleteOptions{})
+			Expect(err).ToNot(HaveOccurred())
+		}()
+
+		By("Creating test-pod-version-0")
+		testPodA := GetTestPod(TestingNamespace)
+		testPodA.SetName("test-pod-version-0")
+		testPodA.Unstructured, err = client.Resource("pods").Namespace(testPodA.GetNamespace()).Create(context.Background(), testPodA.Unstructured, metav1.CreateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		defer func() {
+			err := client.Resource("pods").Namespace(testPodA.GetNamespace()).Delete(context.Background(), testPodA.GetName(), metav1.DeleteOptions{})
+			Expect(err).ToNot(HaveOccurred())
+		}()
+
+		By("Asserting that test-pod-version-0 was labeled with mutation-version: 0")
+		value, ok, err := unstructured.NestedString(testPodA.Object, "metadata", amDef.metadata.kind, amDef.metadata.key)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(ok).To(BeTrue())
+		Expect(value).To(Equal("0"))
+
+		By("Updating assignMetadata to mutation-version: 1")
+		err = unstructured.SetNestedField(am.Object, "1", "spec", "parameters", "assign", "value")
```

Comment:
you know the object type you are working with here so lets use the object and not the `unstructured`

Reply:
the gatekeeper repo does not support this.